packages/common/src/pivot/pivotQueryResults.ts

```diff
@@ -375,6 +376,145 @@ const combinedRetrofit = (
     return data;
 };
 
+const getHeaderValueTypes = ({
```

Comment:
Extracted some logic from the existing `pivotQueryResults` to util functions so they could be reused in the new function.